src/services/dimension-processor.ts

```diff
@@ -533,18 +582,157 @@ export const validateDateDimension = async (
   } catch (error) {
     logger.error(error, `Something unexpected went wrong trying to validate the data`);
     return viewErrorGenerators(500, dataset.id, 'patch', 'errors.dimension_validation.unknown_error', {});
+  } finally {
+    void cubeDB.release();
   }
   return undefined;
 };
 
+interface DateDimensionCreationError {
+  status: number;
+  message: string;
+  dataLength: number;
+  error: Error;
+}
+
+export const createDateDimensionLookup = async (
+  schemaId: string,
+  datasetId: string,
+  dimensionTableName: string,
+  factTableColumn: FactTableColumn,
+  extractor: DateExtractor
+): Promise<{ startDate: Date; endDate: Date; lookupTable: LookupTable }> => {
+  const tableName = 'fact_table';
+  const dataQuery = pgformat(
+    'SELECT DISTINCT %I as date_data FROM %I.%I;',
+    factTableColumn.columnName,
+    schemaId,
+    tableName
+  );
+
+  const getDateDataQueryRunner = dbManager.getCubeDataSource().createQueryRunner();
+  let dateData: { date_data: string }[];
+  try {
+    dateData = await getDateDataQueryRunner.query(dataQuery);
+  } catch (error) {
+    logger.error(error, 'Unable to get date data from the fact table.');
+    throw {
+      status: 500,
+      message: 'errors.dimension_validation.lookup_table_loading_failed',
+      error,
+      dataLength: -1
+    };
+  } finally {
+    void getDateDataQueryRunner.release();
+  }
+
+  let dateDimensionTable: DateReferenceDataItem[] = [];
+  try {
+    dateDimensionTable = dateDimensionReferenceTableCreator(extractor, dateData);
+  } catch (error) {
+    logger.error(error, `Something went wrong trying to create the date reference table`);
+    throw {
+      status: 400,
+      message: 'errors.dimension.invalid_date_format',
+      error,
+      dataLength: dateData.length
+    };
+  }
+
+  const csv = Buffer.from(
+    stringify(
+      dateDimensionTable.map((row) => {
+        return {
+          reference: row.dateCode,
+          language: row.lang,
+          description: row.description,
+          start_date: row.start,
+          end_date: row.end,
+          type: row.type,
+          sort_order: row.end.getTime(),
+          hierarchy: row.hierarchy
+        };
+      }),
+      { bom: true, header: true, quoted_string: true }
+    )
+  );
+
+  const lookupTable = new LookupTable();
+  lookupTable.id = crypto.randomUUID();
+  lookupTable.isStatsWales2Format = false;
+  lookupTable.uploadedAt = new Date();
+  lookupTable.originalFilename = `${lookupTable.id}_${factTableColumn.columnName}_date_tbl.csv`;
+  lookupTable.filename = `${lookupTable.id}_${factTableColumn.columnName}_date_tbl.csv`;
+  lookupTable.fileType = FileType.Csv;
+  lookupTable.mimeType = 'text/csv';
+  const hash = crypto.createHash('sha256');
+  hash.update(csv);
+  lookupTable.hash = hash.digest('hex');
+  await lookupTable.save();
+
+  void getFileService().saveBuffer(lookupTable.originalFilename, datasetId, csv);
```

Comment:
What's the purpose of this csv? Is it a temporary store for something we use in the cube db later on? Can we add a comment or a debug log message just to remind future us please.

Reply:
Its for debugging mostly... It's a full copy of the table generated by the date matching functions.  It has the advantage of turning whats produced by the date matcher into just "another lookup table".  If we had to rebuild everything we have a copy of the date lookup without having to try to process the cube first.